src/libtest/Cargo.toml

```diff
@@ -10,8 +10,22 @@ path = "lib.rs"
 crate-type = ["dylib", "rlib"]
 
 [dependencies]
-getopts = "0.2.19"
+getopts = { version = "0.2.21", features = ['rustc-dep-of-std'] }
```

Comment:
So this should allow libtest to easily depend on dependencies from crates.io as long as they have a:

```toml
#[dependencies]
std = { version = "1.0", package = "rustc-std-workspace-std", optional = true }
core = { version = "1.0", package = "rustc-std-workspace-core", optional = true }

#[feature]
rustc-dep-of-std = ['std', 'core']
```

in their Cargo.toml ?

Reply:
Mostly, i think everything in their dependency tree needs to have this.

Reply:
see also #63725.

Reply:
@gnzlbg yes, similar to how `getopts` [now looks](https://github.com/rust-lang-nursery/getopts/blob/ded6c8edef7a588f23f119c580622387d9350e25/Cargo.toml#L16-L25)